analyzere/base_resources.py

```diff
@@ -48,6 +52,35 @@ def __deepcopy__(self, memo):
             return copy.deepcopy(self.__wrapped__, memo)
         return Reference(self._href)
 
+    def __getattribute__(self, name):
+        # Check for class methods and attributes before intercepting Proxied
+        # methods and attributes.
+        if hasattr(Reference, name):
+            return super(Reference, self).__getattribute__(name)
+
+        attr = Proxy.__getattribute__(self, name)
+        # Intercept all proxied attributes and methods and attempt to update
+        # the Reference ._id and ._href attributes.
+
+        def update_ref(item):
+            try:
+                id_ = item.id
+                href_ = urljoin(analyzere.base_url, item._get_path(id_))
+                super(Reference, self).__setattr__('_id', id_)
+                super(Reference, self).__setattr__('_href', href_)
+            except AttributeError:
+                pass
+
+        if hasattr(attr, '__call__'):
+            def newfunc(*args, **kwargs):
+                retval = attr(*args, **kwargs)
+                update_ref(retval)
+                return retval
+            return newfunc
+        else:
+            update_ref(Proxy)
```

Comment:
Will this ever work? This passes in the class `Proxy`, but then we call `item.id` in `update_ref()`. I would assume that it hits the `try...except` here, because the class itself will not have the attribute `id` but I may be missing something. Python objects always confuse me.